File: internal/handler/team.go
package handler

import (
	"github.com/gin-gonic/gin"
	"github.com/justaboyhai-wq/fmind/internal/types"
	"github.com/justaboyhai-wq/fmind/internal/types/interfaces"
	"net/http"
)

type TeamHandler struct{ service interfaces.TeamService }

func NewTeamHandler(s interfaces.TeamService) *TeamHandler { return &TeamHandler{service: s} }
func (h *TeamHandler) ListDepartments(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	v, e := h.service.ListDepartments(c, t)
	if e != nil {
		c.Error(e)
		return
	}
	c.JSON(http.StatusOK, gin.H{"success": true, "data": v})
}
func (h *TeamHandler) CreateDepartment(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	var x struct {
		Name string `json:"name"`
		Code string `json:"code"`
	}
	if c.ShouldBindJSON(&x) != nil {
		c.JSON(400, gin.H{"error": "invalid request"})
		return
	}
	v, e := h.service.CreateDepartment(c, t, x.Name, x.Code)
	if e != nil {
		c.Error(e)
		return
	}
	c.JSON(http.StatusCreated, gin.H{"success": true, "data": v})
}
func (h *TeamHandler) ListTeams(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	v, e := h.service.ListTeams(c, t, c.Query("department_id"))
	if e != nil {
		c.Error(e)
		return
	}
	c.JSON(http.StatusOK, gin.H{"success": true, "data": v})
}
func (h *TeamHandler) CreateTeam(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	var x struct {
		DepartmentID string `json:"department_id"`
		Name         string `json:"name"`
		Code         string `json:"code"`
	}
	if c.ShouldBindJSON(&x) != nil {
		c.JSON(400, gin.H{"error": "invalid request"})
		return
	}
	v, e := h.service.CreateTeam(c, t, x.DepartmentID, x.Name, x.Code, "")
	if e != nil {
		c.Error(e)
		return
	}
	c.JSON(http.StatusCreated, gin.H{"success": true, "data": v})
}
func (h *TeamHandler) UpdateTeam(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	var x types.Team
	if c.ShouldBindJSON(&x) != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": "invalid request"})
		return
	}
	x.ID = c.Param("id")
	x.TenantID = t
	if e := h.service.UpdateTeam(c, t, &x); e != nil {
		c.Error(e)
		return
	}
	c.JSON(http.StatusOK, gin.H{"success": true})
}
func (h *TeamHandler) DeleteTeam(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	if e := h.service.DeleteTeam(c, t, c.Param("id")); e != nil {
		c.Error(e)
		return
	}
	c.Status(http.StatusNoContent)
}
func (h *TeamHandler) ListMembers(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	v, e := h.service.ListMembers(c, t, c.Param("id"))
	if e != nil {
		c.Error(e)
		return
	}
	c.JSON(200, gin.H{"success": true, "data": v})
}
func (h *TeamHandler) AddMember(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	var x types.TeamMember
	if c.ShouldBindJSON(&x) != nil {
		c.JSON(400, gin.H{"error": "invalid request"})
		return
	}
	x.TeamID = c.Param("id")
	x.TenantID = t
	e := h.service.AddMember(c, t, &x)
	if e != nil {
		c.Error(e)
		return
	}
	c.JSON(http.StatusCreated, gin.H{"success": true})
}
func (h *TeamHandler) RemoveMember(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	e := h.service.RemoveMember(c, t, c.Param("id"), c.Param("user_id"))
	if e != nil {
		c.Error(e)
		return
	}
	c.Status(http.StatusNoContent)
}
func (h *TeamHandler) ListAgents(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	v, e := h.service.ListAgents(c, t, c.Param("id"))
	if e != nil {
		c.Error(e)
		return
	}
	c.JSON(http.StatusOK, gin.H{"success": true, "data": v})
}
func (h *TeamHandler) AddAgent(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	var x types.TeamAgent
	if c.ShouldBindJSON(&x) != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": "invalid request"})
		return
	}
	x.TeamID = c.Param("id")
	x.TenantID = t
	if e := h.service.AddAgent(c, t, &x); e != nil {
		c.Error(e)
		return
	}
	c.JSON(http.StatusCreated, gin.H{"success": true})
}
func (h *TeamHandler) RemoveAgent(c *gin.Context) {
	t := types.MustTenantIDFromContext(c.Request.Context())
	if e := h.service.RemoveAgent(c, t, c.Param("id"), c.Param("agent_id")); e != nil {
		c.Error(e)
		return
	}
	c.Status(http.StatusNoContent)
}

File: frontend/src/api/organization/team.ts
import { get, post, put, del } from '@/utils/request'

export interface Department {
  id: string
  tenant_id: number
  name: string
  code: string
  status: string
}

export interface Team {
  id: string
  tenant_id: number
  department_id: string
  name: string
  code: string
  status: string
}

export interface TeamMember {
  id?: number
  team_id: string
  tenant_id: number
  user_id: string
  role?: string
  status?: string
}

export interface TeamAgent {
  id?: number
  team_id: string
  tenant_id: number
  agent_id: string
  status?: string
}

type ApiList<T> = { success: boolean; data?: T[]; message?: string }

export const listDepartments = () => get<ApiList<Department>>('/api/v1/organization/departments')
export const createDepartment = (data: Pick<Department, 'name' | 'code'>) =>
  post<{ success: boolean; data?: Department }>('/api/v1/organization/departments', data)

export const listTeams = (departmentId?: string) => {
  const query = departmentId ? `?department_id=${encodeURIComponent(departmentId)}` : ''
  return get<ApiList<Team>>(`/api/v1/organization/teams${query}`)
}
export const createTeam = (data: { department_id: string; name: string; code: string }) =>
  post<{ success: boolean; data?: Team }>('/api/v1/organization/teams', data)
export const updateTeam = (id: string, data: Pick<Team, 'department_id' | 'name' | 'code'>) =>
  put<{ success: boolean }>(`/api/v1/organization/teams/${encodeURIComponent(id)}`, data)
export const deleteTeam = (id: string) =>
  del<{ success: boolean }>(`/api/v1/organization/teams/${encodeURIComponent(id)}`)

export const listTeamMembers = (teamId: string) =>
  get<ApiList<TeamMember>>(`/api/v1/organization/teams/${encodeURIComponent(teamId)}/members`)
export const addTeamMember = (teamId: string, userId: string) =>
  post<{ success: boolean }>(`/api/v1/organization/teams/${encodeURIComponent(teamId)}/members`, { user_id: userId })
export const removeTeamMember = (teamId: string, userId: string) =>
  del<{ success: boolean }>(`/api/v1/organization/teams/${encodeURIComponent(teamId)}/members/${encodeURIComponent(userId)}`)

export const listTeamAgents = (teamId: string) =>
  get<ApiList<TeamAgent>>(`/api/v1/organization/teams/${encodeURIComponent(teamId)}/agents`)
export const addTeamAgent = (teamId: string, agentId: string) =>
  post<{ success: boolean }>(`/api/v1/organization/teams/${encodeURIComponent(teamId)}/agents`, { agent_id: agentId })
export const removeTeamAgent = (teamId: string, agentId: string) =>
  del<{ success: boolean }>(`/api/v1/organization/teams/${encodeURIComponent(teamId)}/agents/${encodeURIComponent(agentId)}`)
